### ISSUE TITLE
Improve docs

### ISSUE THREAD

**Issue by crunchypi** [crunchypi/iox#5]
Baseline docs were greatly improved with #1, but things could be even better. Additionally, the go playground links for batching ( #2 ) should be updated.

Things to do (readme):
1. Update go playground for the batching stuff
2. Move errors section up above core interfaces, this should be highlighted more.
3. Convert permutations section into a details button. Maybe even remove comments in the code block.
4. Remove impl pattern section. Instead, add a new details section below 4 with a list of all impl structs. Each list element should be a go playground link.
5. Merge sections Constructors and Modifiers? They're technically all constructors. However, one could argue that it's nice to have a distinction between "From" and "With" funcs, i.e `New<Reader|Writer>From<Something>` and `New<Reader|Writer>With<Something>`. This separation may even be more helpful after #3 and #4 .

Things to do (code):
6. Update go playground links, same as 1 above
7. Add go playground links for impl structs, same as 4 above
